modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ScriptProcessor.java

```diff
@@ -87,66 +85,27 @@ Script getScript() {
-        private final DeprecationLogger deprecationLogger = new DeprecationLogger(logger);
-
         private final ScriptService scriptService;
 
         public Factory(ScriptService scriptService) {
             this.scriptService = scriptService;
         }
 
         @Override
-        @SuppressWarnings("unchecked")
         public ScriptProcessor create(Map<String, Processor.Factory> registry, String processorTag,
                                       Map<String, Object> config) throws Exception {
-            String lang = readOptionalStringProperty(TYPE, processorTag, config, "lang");
-            String source = readOptionalStringProperty(TYPE, processorTag, config, "source");
-            String id = readOptionalStringProperty(TYPE, processorTag, config, "id");
-            Map<String, ?> params = readOptionalMap(TYPE, processorTag, config, "params");
-
-            if (source == null) {
-                source = readOptionalStringProperty(TYPE, processorTag, config, "inline");
-                if (source != null) {
-                    deprecationLogger.deprecated("Specifying script source with [inline] is deprecated, use [source] instead.");
-                }
-            }
-
-            boolean containsNoScript = !hasLength(id) && !hasLength(source);
-            if (containsNoScript) {
-                throw newConfigurationException(TYPE, processorTag, null, "Need [id] or [source] parameter to refer to scripts");
-            }
-
-            boolean moreThanOneConfigured = Strings.hasLength(id) && Strings.hasLength(source);
-            if (moreThanOneConfigured) {
-                throw newConfigurationException(TYPE, processorTag, null, "Only one of [id] or [source] may be configured");
-            }
-
-            if (lang == null) {
-                lang = Script.DEFAULT_SCRIPT_LANG;
-            }
+            XContentBuilder builder = XContentBuilder.builder(JsonXContent.jsonXContent).map(config);
+            JsonXContentParser parser = new JsonXContentParser(NamedXContentRegistry.EMPTY,
+                JSON_FACTORY.createParser(builder.bytes().streamInput()));
+            Script script = Script.parse(parser);
 
-            if (params == null) {
-                params = emptyMap();
-            }
-
-            final Script script;
-            String scriptPropertyUsed;
-            if (Strings.hasLength(source)) {
-                script = new Script(INLINE, lang, source, (Map<String, Object>)params);
-                scriptPropertyUsed = "source";
-            } else if (Strings.hasLength(id)) {
-                script = new Script(STORED, null, id, (Map<String, Object>)params);
-                scriptPropertyUsed = "id";
-            } else {
-                throw newConfigurationException(TYPE, processorTag, null, "Could not initialize script");
-            }
+            Arrays.asList("id", "source", "inline", "lang", "params", "options").forEach(config::remove);
```

Comment:
I don't think this is needed since we are not verifying any input-properties of the script configuration, and letting Script.parse do all the handling and validation.